### DISCUSSION TITLE
PS1.7. Extra variables in email template

### DISCUSSION THREAD

**Discussion by jjmontalban** [PrestaShop/PrestaShop#35316] [Q&A]
I have an ps_emailalerts override to add some variables in new order admin email. I use another module to calculate some variables and insert in $templat_vars array. Debuggin values are correct and when I call Mail::send values are correct. And in new_order.html template smarty varaibles are correct.

We test this override in local (wamp server with clone of production shop, same PHP version, same config, etc.) and works. But if apply override ijn production doesnt work. recargo value is always 0. But doesnt work only {recargo}. Rest of variables ({iva} & {base_imponible}) display correctly ?¿ I have tried:

Change name of variable {recargo}. May be is used in another site. But not.

apply $recargo value directly without $contextLocale->formatPrice but display 0 too.

Debugging all and Mail function. Always display correct value. But when email come {recargo} display 0.

          class Ps_EmailAlertsOverride extends Ps_EmailAlerts
          {

                public function hookActionValidateOrder($params){
             ...
               $recargo = 0;
                 if (Configuration::get('NOHRE_clientesConRecargo') && in_array($id_customer, explode('|',                         
          Configuration::get('NOHRE_clientesConRecargo')))){
                  $recargoEquivalenciaModule = Module::getInstanceByName('recargoequivalencia');
                  $cart = new Cart($params['cart']->id);
                  $taxBreakdown = $recargoEquivalenciaModule->getTaxBreakdownPublic($cart);
                  foreach ($taxBreakdown as $taxRates) {
                  $recargo += $taxRates['total_surcharge'];
               }
             }

     $base_imponible = $total_products + $order->total_shipping_tax_excl - $order->total_discounts_tax_excl;

     // Filling-in vars for email
      $template_vars = [
      ...
      '{recargo}' =>  $contextLocale->formatPrice($recargo, $currency->iso_code),
      '{iva}' => $contextLocale->formatPrice($order->total_paid_tax_incl - $order->total_paid_tax_excl - $recargo, $currency->iso_code),
      '{base_imponible}' => $contextLocale->formatPrice($base_imponible, $currency->iso_code),

      ];

      ... 
       if ($dir_mail) {
           Mail::send(
               $mail_id_lang,
               'new_order',
               $this->trans(
                  'Nuevo pedido : #%d',
                   [
                       $order->id,
                       $order->reference,
                   ],
                   'Emails.Subject',
                   $locale),
               $template_vars,
               $merchant_mail,
               null,
               $configuration['PS_SHOP_EMAIL'],
               $configuration['PS_SHOP_NAME'],
               null,
               null,
               $dir_mail,
               false,
               $id_shop
           );
       }
   }

   }
}

I dont understand why fail only this variable. Repeat. Works in local and doesnt work in production. And only fail this variable!

Any idea? Kind regards...

**Comment by kpodemski** [comment]
If you change this:

` '{recargo}' =>  $contextLocale->formatPrice($recargo, $currency->iso_code),`

to, for instance, this:

`  '{recargo}' => 'test_value',`

does it still display `0` in the email? if not, maybe the problem is somewhere there:

```
 if (Configuration::get('NOHRE_clientesConRecargo') && in_array($id_customer, explode('|',                         
      Configuration::get('NOHRE_clientesConRecargo')))){
```

**Reply by jjmontalban** [reply]
Hi

Thank for you response

If I insert a string result is the same.

The conditional is only a filter to get customers with this type of tax (recargo) but the if works fine. Only get these customers. And $recargo calculate fine. 

I think that the problem may be is in the email sending.
